plugin/core/sessions.py

```diff
@@ -368,6 +370,9 @@ def _handle_initialize_result(self, result: Any) -> None:
         self.on_request("client/registerCapability", self._handle_register_capability)
         self.on_request("client/unregisterCapability", self._handle_unregister_capability)
 
+        if self.config.settings is not None:
```

Comment:
`settings` are never none because we default to empty Dict. Can you just do an `if` check without explicitly comparing with `None`? Then even Jedi should be happy (maybe).

Reply:
Well, jedi won't be happy because it also wants `settings`. It actively requests them with `workspace/configuration` request instead.

Anyway, it doesn't matter much because it keeps on jugging along.

The spec is perhaps too vague about this:
- Must editor notify `workspace/didChangeConfiguration`, always at startup?
- If so, what is the point of registering for `workspace/didChangeConfiguration`?
- If editor always notifies `workspace/didChangeConfiguration`, what is the point of `workspace/configuration` request?

Reply:
Current logic also breaks `LSP-vue`. It expects not the get that call at that point apparently.

Reply:
Maybe we should:
 - notify at starting if there are non-empty settings
 - notify on configuration change only if server did dynamic registration (of course we won't do that right now anyway).